### DISCUSSION TITLE
Add the Ruby on Rails template?

### DISCUSSION THREAD

**Discussion by jcbhmr** [devcontainers-community/templates#26] [Ideas]
![image](https://user-images.githubusercontent.com/61068799/235503794-7f607746-5659-4ccb-9f5c-29d53a9cbafb.png)

is this too similar to a stock ruby image? https://github.com/devcontainers/images/tree/main/src/ruby
basically is it like vue was #17 or is it distinct enough with enough custom "stuff" to add that it warrants a tempalte?

thoughts @eitsupi and anyone else?

**Comment by eitsupi** [comment]
Thanks for opening this.

Perhaps this is the only difference in this one line?
https://github.com/microsoft/vscode-dev-containers/blob/5a084a93b0736ea86395ac99019a5b72a00b6341/containers/ruby-rails/.devcontainer/Dockerfile#L6

As with vue, it would make more sense not to maintain it as a stand-alone template for now, I think.

**Reply by jcbhmr** [reply]
i agree.

for anyone else who stumbles across this, you can do:

```jsonc
{
  "image": "mcr.microsoft.com/devcontainers/ruby",
  "features": {
    // Install Node.js if needed for your frontend tools
    "ghcr.io/devcontainers/features/node": {}
  },
  "containerEnv": {
    // https://stackoverflow.com/q/53878453
    "RAILS_DEVELOPMENT_HOSTS": ".githubpreview.dev,.app.github.dev"
  },
  "updateContentCommand": "gem install rails webdrivers"
}
```

and that should achieve: ruby with rails installed and node.js (if you add the node feature)
you can add more stuff in the `features: {}` object if needed!

(unsure if containerEnv or remoteEnv is best here https://github.com/devcontainers/spec/issues/233)
